crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -57,11 +59,11 @@ def validate_proposal_after_save proposal
     proposal["attributes"]["provisioner"]["access_keys"].strip.split("\n").each do |key|
       key.strip!
 
-      next if key.empty?
+      next if key.empty? || key[0] == "#"
 
-      method, public_key, comment = key.split(" ")
-      unless ["ssh-rsa", "ssh-ed25519", "ssh-dss"].include? method
-        validation_error("SSH key \"#{key}\" should start with a supported \"ssh-*\" method.")
+      _, method, public_key, comment = split_key_line(key)
+      unless valid_key_types.include? method
```

Comment:
So `valid_key_types` is accessible just like this, no module prefix? (Pardon my missing Ruby knowledge)

Reply:
We `include` the module into this class so yes, it is thrown directly into the class namespace. But (with my limited ruby knowledge) I had to spend quite some time to figure out how to do it "properly" :o)